### ISSUE TITLE
Bug with Universal: MatRipple 'window is not defined' - on first SSR Request

### ISSUE THREAD

**Issue by CanKattwinkel** [angular/material2#9684]
#### Bug, feature request, or proposal:

* **BUG**

#### What is the current behavior?

My SSR application is running into a `window is not defined` error on rendering. I can narrow it down by removing all usages of mat-ripple. By usage I mean **explicit usage** of the mat-ripple directive. Using a mat-button does **not** lead to the error.

The Error occurs as sonn as I import MatRippleModule and use it in my HTML markup and import the MatRippleModule Module. All this happens in a lazy loaded feature module (/foo). On the first request (not server startup) the attached error is thrown in the NodeJS console. Then the SSR NodeJS is no longer reactive for around 2 minutes. After that period I can access the site. And even if i access /foo directly to trigger the SSR, the ripple directive works.

So i can see the class has been added correctly by SSR. There is also no more runtime error on the console. This means the error only happens once at application startup?

Generated Markup:
```
  <div _ngcontent-c26="" class="request item mat-ripple" mat-ripple="">
      Some Text
        <mat-icon _ngcontent-c26="" class="mat-icon material-icons" role="img" style="font-weight:300;visibility:hidden;" aria-hidden="true">
            check
        </mat-icon>
    </div>
```

SRC:
```
<div class="request item"
         mat-ripple
         (click)="..."
         [class.active]="..">
        Some Text
        <mat-icon>
            check
        </mat-icon>
    </div>
```

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
```
Angular CLI: 1.6.5
Node: 9.4.0
OS: linux x64
Angular: 5.2.1
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

@angular/cdk: 5.0.0-rc.2
@angular/cli: 1.6.5
@angular/material: 5.1.0
@angular-devkit/build-optimizer: 0.0.41
@angular-devkit/core: 0.0.28
@angular-devkit/schematics: 0.0.51
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.5
@schematics/angular: 0.1.16
typescript: 2.5.3
webpack: 3.10.0

```

#### Is there anything else we should know?
Full Stack:
```

ERROR ReferenceError: window is not defined
    at supportsPassiveEventListeners (/home/can/projects/projectA/client/dist-client/server.js:36162:13)
    at new RippleRenderer (/home/can/projects/projectA/client/dist-client/server.js:29269:122)
    at new MatRipple (/home/can/projects/projectA/client/dist-client/server.js:29542:32)
    at createClass (/home/can/projects/projectA/client/dist-client/server.js:13261:20)
    at createDirectiveInstance (/home/can/projects/projectA/client/dist-client/server.js:13096:37)
    at createViewNodes (/home/can/projects/projectA/client/dist-client/server.js:14554:53)
    at callViewAction (/home/can/projects/projectA/client/dist-client/server.js:14988:13)
    at execComponentViewsAction (/home/can/projects/projectA/client/dist-client/server.js:14897:13)
    at createViewNodes (/home/can/projects/projectA/client/dist-client/server.js:14582:5)
    at callViewAction (/home/can/projects/projectA/client/dist-client/server.js:14988:13)
Unhandled Promise rejection: window is not defined ; Zone: <root> ; Task: Promise.then ; Value: ReferenceError: window is not defined
    at supportsPassiveEventListeners (/home/can/projects/projectA/client/dist-client/server.js:36162:13)
    at new RippleRenderer (/home/can/projects/projectA/client/dist-client/server.js:29269:122)
    at new MatRipple (/home/can/projects/projectA/client/dist-client/server.js:29542:32)
    at createClass (/home/can/projects/projectA/client/dist-client/server.js:13261:20)
    at createDirectiveInstance (/home/can/projects/projectA/client/dist-client/server.js:13096:37)
    at createViewNodes (/home/can/projects/projectA/client/dist-client/server.js:14554:53)
    at callViewAction (/home/can/projects/projectA/client/dist-client/server.js:14988:13)
    at execComponentViewsAction (/home/can/projects/projectA/client/dist-client/server.js:14897:13)
    at createViewNodes (/home/can/projects/projectA/client/dist-client/server.js:14582:5)
    at callViewAction (/home/can/projects/projectA/client/dist-client/server.js:14988:13) ReferenceError: window is not defined
    at supportsPassiveEventListeners (/home/can/projects/projectA/client/dist-client/server.js:36162:13)
    at new RippleRenderer (/home/can/projects/projectA/client/dist-client/server.js:29269:122)
    at new MatRipple (/home/can/projects/projectA/client/dist-client/server.js:29542:32)
    at createClass (/home/can/projects/projectA/client/dist-client/server.js:13261:20)
    at createDirectiveInstance (/home/can/projects/projectA/client/dist-client/server.js:13096:37)
    at createViewNodes (/home/can/projects/projectA/client/dist-client/server.js:14554:53)
    at callViewAction (/home/can/projects/projectA/client/dist-client/server.js:14988:13)
    at execComponentViewsAction (/home/can/projects/projectA/client/dist-client/server.js:14897:13)
    at createViewNodes (/home/can/projects/projectA/client/dist-client/server.js:14582:5)
    at callViewAction (/home/can/projects/projectA/client/dist-client/server.js:14988:13)


```

**Comment by DevVersion** [comment]
I tried to reproduce the SSR issue, but wasn't able to reproduce the issue within our `universal-app`. The `supportsPassiveEventListeners` method should be SSR-proof since v5.1.0 / https://github.com/angular/material2/commit/12feff70643cfa213ad130504cd2e6c521484b1f.

Since you seem to use `v5.1.0`, I'm wondering why this issue shows up. Do you think you'd be able to share the code part, which the error stack shows? (`dist-client/server.js:36162:13`)

**Comment by CanKattwinkel** [comment]
While i payed a lot of attention to the material version I obviously did not to the cdk version. My package.json was lacking a ^, therefore i got an outdated version of cdk. Also I missed the commit since the message didn't contain any information about server side rendering - if I would have checked. 

So long story short: I can confirm this was fixed by v5.1.0
